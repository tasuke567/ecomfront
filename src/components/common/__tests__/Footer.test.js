// src/components/common/__tests__/Footer.test.js
import React from 'react';
import { render, screen } from '@testing-library/react';
import Footer from '../Footer';

describe('Footer Component', () => {
  test('renders footer with correct background color', () => {
    render(<Footer />);
    const footer = screen.getByTestId('footer');
    expect(footer).toHaveClass('bg-gray-800');
  });

  test('renders footer with correct text color', () => {
    render(<Footer />);
    const footer = screen.getByTestId('footer');
    expect(footer).toHaveClass('text-white');
  });

  test('renders copyright text', () => {
    render(<Footer />);
    expect(screen.getByText(/© 2024 E-Shop\. All rights reserved\./i)).toBeInTheDocument();
  });

  test('renders with correct container styles', () => {
    render(<Footer />);
    const container = screen.getByText(/© 2024/).parentElement;
    expect(container).toHaveClass('container', 'mx-auto', 'text-center');
  });

  test('renders with correct padding', () => {
    render(<Footer />);
    const footer = screen.getByTestId('footer');
    expect(footer).toHaveClass('py-4');
  });

  test('applies correct margin to container', () => {
    render(<Footer />);
    const container = screen.getByText(/© 2024/).parentElement;
    expect(container).toHaveClass('mx-auto');
  });

  test('renders current year in copyright text', () => {
    const currentYear = new Date().getFullYear().toString();
    render(<Footer />);
    expect(screen.getByText(new RegExp(currentYear))).toBeInTheDocument();
  });

  test('maintains fixed structure and styling', () => {
    const { container } = render(<Footer />);
    expect(container.firstChild).toMatchSnapshot();
  });
});