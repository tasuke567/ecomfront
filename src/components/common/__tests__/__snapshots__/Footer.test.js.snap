// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component maintains fixed structure and styling 1`] = `
<footer
  class="bg-gray-800 text-white py-4"
  data-testid="footer"
>
  <div
    class="container mx-auto text-center"
  >
    <p>
      Â© 2024 E-Shop. All rights reserved.
    </p>
  </div>
</footer>
`;
